shared/queue/priority_queue.go

```diff
@@ -149,6 +149,30 @@ func (pq *PriorityQueue) PopByKey(key string) (*Item, error) {
 	return nil, nil
 }
 
+// RetrieveByKey searches the queue for an item with the given key and returns it
+// from the queue if found. Returns nil if not found.
+func (pq *PriorityQueue) RetrieveByKey(key string) (*Item, error) {
```

Comment:
why is there an error in this function signature ? The underlying method never returns an error in any of the paths.